tests/apiclient_test.cc

```diff
@@ -342,6 +342,34 @@ TEST_F(ApiClientTest, InstallModeOstreeOnlyIfJustApps) {
   }
 }
 
+TEST_F(ApiClientTest, InstallWithoutDownload) {
```

Comment:
I would add a positive test that downloads and installs a target by using another instance of liteclient and correspondingly another instance of the installer for this two actions.

Reply:
I believe a test like this would be much more useful if the actual code for the `ComposeAppEngine` is used, instead of `MockAppEngine`. Using `MockAppEngine`, the previous version would still make the test pass.
Where could I place it?

Reply:
The `cli_test` uses the real package manager and app engine.
However, it will require the CLI API extending - adding functions to pull and install separately.

Reply:
I had a hard time trying to make a proper test inside `cli_test.c`. Not only `AkliteClient` has to be changed, but the mocks and actual objects organization inside that file require changes. For example, multiple objects created with `createAkClient` share the same `app_engine`. Just creating separate `app_engine` instances is not that hard, but they would have to share the same and path, which would require more changes. It looks this is too much work for at the moment, for little benefit, so I simply added one test to the api tests, even if it does not test the `ComposeAppEngine` behavior.

Reply:
I understand and that's fine.